jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanReader.java

```diff
@@ -34,19 +34,22 @@ public class BeanReader
     protected final Constructor<?> _defaultCtor;
     protected final Constructor<?> _stringCtor;
     protected final Constructor<?> _longCtor;
+    protected final Constructor<?> _intCtor;
 
     /**
      * Constructors used for deserialization use case
      */
     public BeanReader(Class<?> type, Map<String, BeanPropertyReader> props,
-            Constructor<?> defaultCtor, Constructor<?> stringCtor, Constructor<?> longCtor,
+            Constructor<?> defaultCtor, Constructor<?> stringCtor, Constructor<?> longCtor, Constructor<?> intCtor,
             Set<String> ignorableNames, Map<String, String> aliasMapping)
     {
         super(type);
         _propsByName = props;
         _defaultCtor = defaultCtor;
         _stringCtor = stringCtor;
         _longCtor = longCtor;
+        _intCtor = intCtor;
```

Comment:
I believe propogation of HashMap<FIRST_ARG,CONSTRUCTOR> should be a better alternative, to support future items. instead of adding individually in all the following files.

Reply:
I agree with the general idea (could use `EnumMap` too), but how about going even more OO and have `Constructors` (or whatever its named) container that encapsulates these details?

Reply:
I tried using enum map, but that would require us explicitly manage a list of supported items, with the code getters and setters; a problem not present in HashMap.